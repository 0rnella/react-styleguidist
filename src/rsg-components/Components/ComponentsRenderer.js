import React, { PropTypes } from 'react';

export default function ComponentsRenderer({ components, sections }) {
	return (
		<div>
			{components}
			{sections}
		</div>
	);
}
ComponentsRenderer.propTypes = {
	components: PropTypes.array.isRequired,
	sections: PropTypes.node.isRequired,
};
