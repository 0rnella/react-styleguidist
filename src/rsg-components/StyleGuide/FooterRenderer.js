import React from 'react';
import PropTypes from 'prop-types';
import Markdown from 'rsg-components/Markdown';
import Styled from 'rsg-components/Styled';

const styles = ({ color, fontFamily, fontSize }) => ({
	footer: {
		display: 'block',
		color: color.light,
		fontFamily: fontFamily.base,
		fontSize: fontSize.small,
	},
});

const Footer = ({ classes, homepageUrl }) => (
	<footer className={classes.footer}>
		<Markdown text={`Generated with [React Styleguidist](${homepageUrl})`} />
	</footer>
);

Footer.propTypes = {
	classes: PropTypes.object.isRequired,
	homepageUrl: PropTypes.string.isRequired,
};

export default Styled(styles)(Footer);
