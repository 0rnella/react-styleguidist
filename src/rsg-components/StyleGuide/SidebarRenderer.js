import React from 'react';
import PropTypes from 'prop-types';
import { MdFormatIndentDecrease, MdFormatIndentIncrease } from 'react-icons/lib/md';
import FaFileTextO from 'react-icons/lib/fa/file-text-o';
import ToolbarButton from 'rsg-components/ToolbarButton';
import cx from 'classnames';
import Styled from 'rsg-components/Styled';
import Logo from 'rsg-components/Logo';

const styles = ({ color, sidebarWidth, mq, space }) => ({
	toggleProps: {
		left: space[5] + space[1],
		'& svg': {
			padding: space[0] / 2,
		},
		[mq.small]: {
			display: 'none',
		},
	},
	sidebar: {
		backgroundColor: color.sidebarBackground,
		border: [[color.border, 'solid']],
		borderWidth: [[0, 1, 0, 0]],
		position: 'fixed',
		top: 0,
		left: 0,
		bottom: 0,
		width: sidebarWidth,
		overflow: 'auto',
	},
	logo: {
		padding: space[2],
		borderBottom: [[1, color.border, 'solid']],
	},
});

const Sidebar = ({
	classes,
	title,
	onToggleSidebar,
	sidebarVisible,
	toc,
	toggleAllProps,
	styles,
}) => (
	<div>
		{sidebarVisible ? (
			<aside className={classes.sidebar}>
				<div className={classes.logo}>
					<Logo>{title}</Logo>
				</div>
				{toc}
				<ToolbarButton
					className={styles}
					onClick={onToggleSidebar}
					title="Collapse Sidebar"
					className={styles}
				>
					<MdFormatIndentDecrease />
				</ToolbarButton>
				<ToolbarButton
					onClick={toggleAllProps}
					title="Toggle Focus Mode"
					className={cx(styles, classes.toggleProps)}
				>
					<FaFileTextO />
				</ToolbarButton>
			</aside>
		) : (
			<ToolbarButton title="Expand Sidebar" className={styles} onClick={onToggleSidebar}>
				<MdFormatIndentIncrease />
			</ToolbarButton>
		)}
	</div>
);

Sidebar.propTypes = {
	classes: PropTypes.object.isRequired,
	styles: PropTypes.string.isRequired,
	onToggleSidebar: PropTypes.func.isRequired,
	toggleAllProps: PropTypes.func.isRequired,
	title: PropTypes.string.isRequired,
	sidebarVisible: PropTypes.bool,
	toc: PropTypes.node.isRequired,
};

export default Styled(styles)(Sidebar);
