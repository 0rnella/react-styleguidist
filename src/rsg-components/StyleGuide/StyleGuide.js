import { Component } from 'react';
import Layout from 'rsg-components/Layout';

export default class StyleGuide extends Component {
	render() {
		return (
			<Layout {...this.props}/>
		);
	}
}
