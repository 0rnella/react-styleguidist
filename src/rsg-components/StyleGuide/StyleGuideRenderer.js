import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Styled from 'rsg-components/Styled';
import Footer from './FooterRenderer';
import Sidebar from './SidebarRenderer';

const styles = ({ color, sidebarWidth, mq, space, maxWidth }) => ({
	root: {
		color: color.base,
		backgroundColor: color.baseBackground,
		transition: 'padding-left 200ms ease-in-out',
	},
	hasSidebar: {
		paddingLeft: sidebarWidth,
		[mq.small]: {
			paddingLeft: 0,
		},
	},
	content: {
		maxWidth,
		padding: [[space[2], space[4]]],
		margin: [[0, 'auto']],
		[mq.small]: {
			padding: space[2],
		},
		display: 'block',
	},
});

class StyleGuideRenderer extends Component {
	constructor(props) {
		super(props);
		this.state = {
			sidebarVisible: true,
		};

		this.handleToggleSidebar = this.handleToggleSidebar.bind(this);
	}

	handleToggleSidebar() {
		this.setState({ sidebarVisible: !this.state.sidebarVisible });
	}

	render() {
		const { classes, title, homepageUrl, children, toc, hasSidebar, toggleAllProps } = this.props;
		const { sidebarVisible } = this.state;

		return (
			<div className={cx(classes.root, hasSidebar && sidebarVisible && classes.hasSidebar)}>
				<main className={classes.content}>
					{children}
					<Footer homepageUrl={homepageUrl} />
				</main>
				<Sidebar
					title={title}
					hasSidebar={hasSidebar}
					toc={toc}
					toggleAllProps={toggleAllProps}
					sidebarVisible={sidebarVisible}
					onToggleSidebar={this.handleToggleSidebar}
				/>
			</div>
		);
	}
}

StyleGuideRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	title: PropTypes.string.isRequired,
	homepageUrl: PropTypes.string.isRequired,
	children: PropTypes.node.isRequired,
	toc: PropTypes.node.isRequired,
	hasSidebar: PropTypes.bool,
	toggleAllProps: PropTypes.func,
};

export default Styled(styles)(StyleGuideRenderer);
