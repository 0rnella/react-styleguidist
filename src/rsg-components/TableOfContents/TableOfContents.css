.root {
	composes: reset font from "../../styles/common.css";
	list-style-type: none;
	padding: 15px;
}

.item {
	margin: 0 0 7px;
	padding: 0;
	font-size: 15px;
}

.link {
	composes: link from "../../styles/colors.css";
}
