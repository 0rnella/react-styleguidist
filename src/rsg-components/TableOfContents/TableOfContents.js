import React, { Component, PropTypes } from 'react';

import s from './TableOfContents.css';

class TableOfContents extends Component {

	constructor(props) {
		super(props);

		this.state = {
			searchTerm: '',
			filteredComponents:[]
		};

		this.handleChange = this.handleChange.bind(this);
	}

	searchFilter(value, component) {
		const regex = new RegExp(value.toLowerCase().toString(), 'g');
		return component.name.toLowerCase().match(regex);
	}

	handleChange(e) {

		let components = this.props.components;

		if(e.target.value !== '') {
			components = components.filter(this.searchFilter.bind(null, e.target.value));
		}

		this.setState({
			searchTerm: e.target.value,
			filteredComponents: components
		})
	}

	componentDidMount() {
		this.setState({
			filteredComponents: this.props.components
		 })
	}

	render() {
		const { filteredComponents, searchTerm } = this.state;
		return (
			<div>
				<input
					className={s.find}
					placeholder="Find component..."
					onChange={this.handleChange}
					value={searchTerm} />
				<ul className={s.root}>
					{filteredComponents.map(({name}) => (
						<li className={s.item} key={name}>
							<a className={s.link} href={'#' + name}>{name}</a>
						</li>
					))}
				</ul>
			</div>
		);
	}
};

TableOfContents.propTypes = {
	components: PropTypes.array.isRequired
};

export default TableOfContents;
