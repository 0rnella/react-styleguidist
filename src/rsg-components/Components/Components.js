import React, { Component, PropTypes } from 'react';
import ReactComponent from 'rsg-components/ReactComponent';
import Sections from 'rsg-components/Sections';
import ComponentsRenderer from 'rsg-components/Components/ComponentsRenderer';

export default class Components extends Component {
	static propTypes = {
		highlightTheme: PropTypes.string.isRequired,
		components: PropTypes.array.isRequired,
		sections: PropTypes.array.isRequired,
		sidebar: PropTypes.bool,
	};

	renderComponents() {
		const { highlightTheme, components, sidebar } = this.props;

		return components.map(component => {
			return (
				<ReactComponent
					key={component.filepath}
					highlightTheme={highlightTheme}
					component={component}
					sidebar={sidebar}
				/>
			);
		});
	}

	renderSections() {
		const { highlightTheme, sections, sidebar } = this.props;

		return (
			<Sections
				highlightTheme={highlightTheme}
				sections={sections}
				sidebar={sidebar}
			/>
		);
	}

	render() {
		return (
			<ComponentsRenderer
				components={this.renderComponents()}
				sections={this.renderSections()}
			/>
		);
	}
}
