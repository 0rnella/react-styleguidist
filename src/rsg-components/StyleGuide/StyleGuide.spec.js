import React from 'react';
import Components from '../Components';
import Logo from '../Logo';
import Message from '../Message';
import TableOfContents from '../TableOfContents';
import StyleGuide from './StyleGuide';
import { StyleGuideRenderer } from './StyleGuideRenderer';

const components = [
	{
		name: 'Foo',
		pathLine: 'components/foo.js',
		filepath: 'components/foo.js',
		props: {
			description: 'Foo foo',
		},
	},
	{
		name: 'Bar',
		pathLine: 'components/bar.js',
		filepath: 'components/bar.js',
		props: {
			description: 'Bar bar',
		},
	},
];
const sections = [];
const config = {
	title: 'Hello',
};

it('should render components list', () => {
	const actual = shallow(
		<StyleGuide
			codeKey={1}
			config={config}
			components={components}
			sections={sections}
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('should pass error message instead of components list when there is no components and sections', () => {
	const actual = shallow(
		<StyleGuide
			codeKey={1}
			config={config}
			components={[]}
			sections={[]}
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render logo, table on contents and components', () => {
	const actual = shallow(
		<StyleGuideRenderer
			classes={{}}
			title={config.title}
			components={<Components components={components} sections={sections} />}
			sections={sections}
			toc={<TableOfContents components={components} sections={sections} />}
			homepageUrl="http://react-styleguidist.js.org/"
			sidebar
		/>
	);

	expect(actual).toMatchSnapshot();
});
