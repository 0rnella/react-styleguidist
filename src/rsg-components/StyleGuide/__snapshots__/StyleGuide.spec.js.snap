// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render components list 1`] = `
<Styled(StyleGuide)
  hasSidebar={true}
  homepageUrl="https://react-styleguidist.js.org/"
  title="Hello"
  toc={
    <TableOfContents
      sections={
        Array [
          Object {
            "components": Array [
              Object {
                "filepath": "components/foo.js",
                "name": "Foo",
                "pathLine": "components/foo.js",
                "props": Object {
                  "description": "Foo foo",
                },
              },
              Object {
                "filepath": "components/bar.js",
                "name": "Bar",
                "pathLine": "components/bar.js",
                "props": Object {
                  "description": "Bar bar",
                },
              },
            ],
          },
        ]
      }
    />
  }
  toggleAllProps={[Function]}
>
  <Sections
    depth={1}
    sections={
      Array [
        Object {
          "components": Array [
            Object {
              "filepath": "components/foo.js",
              "name": "Foo",
              "pathLine": "components/foo.js",
              "props": Object {
                "description": "Foo foo",
              },
            },
            Object {
              "filepath": "components/bar.js",
              "name": "Bar",
              "pathLine": "components/bar.js",
              "props": Object {
                "description": "Bar bar",
              },
            },
          ],
        },
      ]
    }
  />
</Styled(StyleGuide)>
`;

exports[`should render welcome screen 1`] = `<Styled(Welcome) />`;
